src/flownet/parameters/_relative_permeability.py

```diff
@@ -165,19 +172,23 @@ def _check_parameters(self) -> Tuple[bool, bool]:
             A tuple of booleans defining whether to generate the SWOF and or SGOF tables.
 
         """
-        # Check if the number of parameters make either sense for SWOF (8), SGOF (8) or both (14)
-        if len(self._parameters) != 8 and len(self._parameters) != 14:
+        if isinstance(self._interpolation_values, pd.DataFrame):
```

Comment:
As `interpolation_values` is will be evaluated as `True` when it is a `pd.DataFrame` and `False` when it is `None`, you can drop the `isinstance()` call.